transports/msmq/delayed-delivery_content_msmqtransport_[2,).partial.md

```diff
@@ -0,0 +1,21 @@
+Because the MSMQ lacks a mechanism for sending delayed messages, the MSMQ transport uses an external store for delayed messages. Messages that are meant to be delivered later (e.g. [saga timeouts](/nservicebus/sagas/timeouts.md) or [delayed retries](/nservicebus/recoverability/configure-delayed-retries.md)) are persisted in this store until there are due. At that point they are fetched from the store and forwarded to their ultimate destinations.
+
+To configure MSMQ to enable delayed message delivery use the following API:
+
+snippet: delayed-delivery
+
+The code above sets up a SQL Server delayed delivery store. It is the only store that ships with the transport. By default it uses a table named after then endpoint with `.delayed` suffix but that name can be customized. 
+
+When using a different database users can provide their own implementations of `IDelayedMessageStore` interface. The delayed delivery handling mechanism has two parts. First, a delayed message is stored using the `Store` method. 
+
+The second part is polling the store for due delayed messages. This is accomplished by periodically calling `FetchNextDueTimeout`. As long as this method returns a message, the mechanism continues to pull more messages. If there are no more due messages, `Next` is invoked to determine how much time is left until the next due timeout. The polling mechanism pauses until that time. If another delayed message is persisted in the meantime, the `Store` method wakes up the polling thread.
+
+When a due delayed message is found, it is forwarded to the destination and then removed from the store using the `Remove` method. In case of an unexpected exception occurs during forwarding the failure is registered using `IncrementFailureCount`. If configured number of retries is exhaused the message is forwarded to the configured `error` queue. Refer to API documentation for details regarding the functionality expected in each method.
```

Comment:
```suggestion
When a due delayed message is returned by `FetchNextDueTimeout`, the message is forwarded to the destination and then removed from the store using the `Remove` method. In case of an unexpected exception during forwarding the failure is registered using `IncrementFailureCount`. If the configured number of retries is exhausted the message is forwarded to the configured `error` queue.
```